core/src/main/java/org/hibernate/ogm/remote/impl/DatabaseIdentifier.java

```diff
@@ -4,7 +4,7 @@
  * License: GNU Lesser General Public License (LGPL), version 2.1 or later
  * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
  */
-package org.hibernate.ogm.datastore.couchdb.util.impl;
```

Comment:
do we really want this implementation to be shared across all dialects?

If you need something like this, I can see the benefit for an interface but I wouldn't assume that this same implementation is good enough for all other dialects (it's not for Infinispan, eg.).